src/librustc/ty/layout.rs

```diff
@@ -511,41 +512,70 @@ pub struct Struct {
     /// If true, the size is exact, otherwise it's only a lower bound.
     pub sized: bool,
 
-    /// Offsets for the first byte of each field.
+    /// Offsets for the first byte of each field, ordered in increasing order.
+    /// The ith element of this vector is not always the ith field of the struct.
     /// FIXME(eddyb) use small vector optimization for the common case.
     pub offsets: Vec<Size>,
```

Comment:
Should I just make it private and have `field_offset` and `num_fields` functions?